vendor/noise_handshake/src/handshake.rs

```diff
@@ -0,0 +1,209 @@
+use futures::*;
+use snow::Session;
+use tokio::io::{AsyncRead, AsyncWrite};
+
+const NOISE_MAX_SIZE: usize = 65535;
+
+#[allow(missing_debug_implementations)]
+pub enum Step {
+    Read {
+        enc_buffer: [u8; NOISE_MAX_SIZE],
+        len: usize,
+    },
+    Write {
+        buffer: [u8; NOISE_MAX_SIZE],
+        written_bytes: usize,
+        len: usize,
+    },
+}
+
+impl Step {
+    fn read() -> Self {
+        Step::Read {
+            enc_buffer: [0u8; NOISE_MAX_SIZE],
+            len: 0,
+        }
+    }
+
+    fn write(noise: &mut Session) -> Self {
+        let mut buffer = [0u8; NOISE_MAX_SIZE];
+        let len = noise
+            .write_message(&[], &mut buffer)
+            .expect("Cannot encode the message");
```

Comment:
❌ With expect you should say why this can never fail: "A zero-length message is cannot be too long".